src/common.c

```diff
@@ -2736,41 +2738,18 @@ int stlink_fwrite_flash(stlink_t *sl, const char* path, stm32_addr_t addr) {
  * @param base option bytes to write
  * @return 0 on success, -ve on failure.
  */
-static int stlink_write_option_bytes_g0x(stlink_t *sl, uint8_t* base, uint32_t len) {
+static int stlink_write_option_bytes_gx(stlink_t *sl, uint8_t* base, stm32_addr_t addr, uint32_t len) {
 
     uint32_t val;
 
-    if (len != 4) {
-        ELOG("Wrong length for writing option bytes, must be 4 is %d\n", len);
-        return -1;
-    }
+    (void) addr;
+    (void) len;
 
-    // Make sure we've loaded the context with the chip details
-    stlink_core_id(sl);
-
-    /* Check if chip is supported and for correct address */
-    if (sl->chip_id != STLINK_CHIPID_STM32_G0_CAT1 &&
-       sl->chip_id != STLINK_CHIPID_STM32_G0_CAT2) {
-        ELOG("Option bytes writing is currently only supported for the STM32G0\n");
+    if (unlock_flash_if(sl)) {
+        ELOG("Flash unlock failed! System reset required to be able to unlock it again!\n");
```

Comment:
Good! :+1: We have seen this in issues before where people did not understand what happened ...